File: Graduation_Project_BE/src/main/java/com/project/app/notification/service/NotificationService.java
package com.project.app.notification.service;

import com.project.app.notification.dto.response.NotificationResponse;
import com.project.app.user.entity.User;

import java.util.List;

public interface NotificationService {
    void createNotification(User user, String title, String message);
    List<NotificationResponse> getUserNotifications(User user);
    long getUnreadCount(User user);
    void markAllAsRead(User user);
}

File: Graduation_Project_FE/shared/api/services/notification.service.ts
import { axiosClient } from "../axiosClient";
import { ENDPOINTS } from "../endpoints";

export interface NotificationResponse {
  id: number;
  title: string;
  message: string;
  type?: string;
  relatedId?: number | null;
  isRead: boolean;
  createdAt: string;
}

export const notificationService = {
  getAll: async () => {
    const response = await axiosClient.get(ENDPOINTS.NOTIFICATION.GET_ALL);
    return response;
  },

  getUnreadCount: async () => {
    const response = await axiosClient.get(ENDPOINTS.NOTIFICATION.UNREAD_COUNT);
    return response;
  },

  readAll: async () => {
    const response = await axiosClient.put(ENDPOINTS.NOTIFICATION.READ_ALL);
    return response;
  },

  delete: async (id: number) => {
    const response = await axiosClient.delete(ENDPOINTS.NOTIFICATION.DELETE(id));
    return response;
  }
};
